Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/E2ETests/E2EAnalysisTests.cs

```diff
@@ -931,12 +931,25 @@ public void AssertExpectedFileList(params string[] fileNames)
                 var actualFilesToAnalyze = ProjectInfo.AssertAnalysisResultExists("FilesToAnalyze");
                 actualFilesToAnalyze.Location.Should().Be(filesToAnalyzeFile.FullPath);
 
+                AssertFileIsUtf8Bom(filesToAnalyzeFile.FullPath);
+
                 if (ProjectConfig != null)
                 {
                     ProjectConfig.FilesToAnalyzePath.Should().Be(filesToAnalyzeFile.FullPath);
                 }
             }
 
+            private void AssertFileIsUtf8Bom(string filePath)
+            {
+                using (var filesToAnalyzeStream = File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.Read))
+                {
+                    filesToAnalyzeStream.Length.Should().BeGreaterThan(3);
+                    var buffer = new byte[3];
+                    filesToAnalyzeStream.Read(buffer, 0, buffer.Length);
```

Comment:
Getting the whole stream length could be costly, as the file size needs to be inspected. And `BeGreaterThan` also requires some content after the preamble. `FilesToAnalyze.txt` could be empty, in theory.

Let's do it like this:
```suggestion
                    var buffer = new byte[3];
                    filesToAnalyzeStream.Read(buffer, 0, buffer.Length).Should().Be(3);
```